frontend-react/src/components/Table/Table.tsx

```diff
@@ -115,69 +147,121 @@ const Table = ({ config, filterManager, cursorManager }: TableProps) => {
-                object[columnConfig.dataAttr]
+                <span>
+                    {columnConfig.valueMap?.get(
+                        object[columnConfig.dataAttr]
+                    ) || object[columnConfig.dataAttr]}
+                </span>
             );
         } else {
-            return object[columnConfig.dataAttr];
+            return <span>{object[columnConfig.dataAttr]}</span>;
         }
     };
 
-    const renderRow = (object: TableRow, columnConfig: ColumnConfig) => {
-        let displayValue = object[columnConfig.dataAttr];
-        // Transforms value if transform function is given
+    const Column = ({
+        rowIndex,
+        colIndex,
+        rowData,
+        columnConfig,
+        editing,
+    }: ColumnProps) => {
+        const tableData = (child: ReactNode) => (
+            <td key={`${rowIndex}:${colIndex}`}>{child}</td>
+        );
+
         if (columnConfig.transform) {
-            displayValue = columnConfig.transform(displayValue);
+            rowData[columnConfig.dataAttr] = columnConfig.transform(
+                rowData[columnConfig.dataAttr]
+            );
         }
 
         if (columnConfig.link) {
             // Render column value as NavLink
-            return (
+            return tableData(
                 <NavLink
                     className="usa-link"
                     to={`${columnConfig?.linkBasePath || ""}${
-                        object[columnConfig?.linkAttr || columnConfig.dataAttr]
+                        rowData[columnConfig?.linkAttr || columnConfig.dataAttr]
                     }`}
                 >
                     {columnConfig.valueMap
-                        ? showMappedValue(columnConfig, object)
-                        : displayValue}
+                        ? showMappedValue(columnConfig, rowData)
+                        : rowData[columnConfig.dataAttr]}
                 </NavLink>
             );
         } else if (columnConfig.actionable) {
             const { action, param } = columnConfig.actionable;
             const doAction = () => {
-                if (param) return action(object[param]);
+                if (param) return action(rowData[param]);
                 return action();
             };
-            return (
+            return tableData(
                 <button
                     className="usa-link bg-transparent border-transparent"
                     onClick={() => doAction()}
                 >
-                    {displayValue}
+                    {rowData[columnConfig.dataAttr]}
                 </button>
             );
         } else {
+            if (editing && columnConfig.editable)
+                return tableData(
+                    <input
+                        className="usa-input"
+                        onChange={(event) =>
+                            (rowData[columnConfig.dataAttr] =
+                                event.target.value)
```

Comment:
This will change the object attribute at the level you pass in the item array, so no callback functions required! :)

Reply:
This also ensures editing a valueMapped column means you're editing the pre-mapped value, not the mapped outcome value.

Reply:
These are useful things for future developers to understand when looking at this code. Maybe add these points as comments into the code itself so it lives on?